src/lib/components/NglMoleculeViewer.react.js

```diff
@@ -0,0 +1,274 @@
+        });
+        stage
+            .loadFile(stringBlob, {ext: data.ext, defaultRepresentation: false})
+            .then(stageObj => {
+                stageObj.name = data.filename;
+                this.showStructure(
+                    stageObj,
+                    data.chain,
+                    data.color,
+                    xOffset,
+                    stage
+                );
+
+                this.setState(state => ({
+                    structuresList: state.structuresList.concat([
+                        data.filename,
+                    ]),
+                }));
+            });
+    }
+
+    render() {
+        const {id} = this.props;
+        return (
+            <div>
+                <div id={id} />
+            </div>
+        );
+    }
+}
+
+const defaultViewportStyle = {
+    width: '500px',
+    height: '500px',
+};
+
+const defaultStageParameters = {
+    quality: 'medium',
+    backgroundColor: 'white',
+    cameraType: 'perspective',
+};
+
+const defaultData = [
+    {
+        selectedValue: 'placeholder',
+        chain: 'ALL',
+        color: 'red',
+        filename: 'placeholder',
+        ext: '',
+        config: {
+            input: '',
+            type: 'text/plain',
+        },
+    },
+];
+
+NglMoleculeViewer.defaultProps = {
+    data: defaultData,
+    viewportStyle: defaultViewportStyle,
+    stageParameters: defaultStageParameters,
+};
+
+NglMoleculeViewer.propTypes = {
+    /**
+     * The ID of this component, used to identify dash components in callbacks.
+     * The ID needs to be unique across all of the components in an app.
+     */
+    id: PropTypes.string,
+
+    /**
+     * The height and the width (in px) of the container
+     * in which the molecules will be displayed.
+     * It should be in JSON format.
+     */
+    viewportStyle: PropTypes.exact({
+        width: PropTypes.string,
+        height: PropTypes.string,
+    }),
+
+    /**
+     * Parameters (in JSON format) for the stage object of ngl.
+     * Currently implemented are render quality, background color and camera type
+     * quality: auto, low, medium, high (default: auto)
+     * backgroundColor: white / black (default: white)
+     * cameraType: perspective / orthographic (default: perspective)
+     */
+    stageParameters: PropTypes.exact({
+        quality: PropTypes.string,
+        backgroundColor: PropTypes.string,
+        cameraType: PropTypes.string,
+    }),
+
+    /**
+     * Variable which defines how many molecules should be shown and/or which chain
+     * The following format needs to be used:
+     * pdbID1.chain_pdbID2.chain
+     * . indicates that only one chain should be shown
+     *  _ indicates that more than one protein should be shown
+     */
+    pdbString: PropTypes.string,
+
+    /**
+     * The data (in JSON format) that will be used to display the molecule
+     * selectedValue: pdbString
+     * color: color in hex format
+     * filename: name of the used pdb/cif file
```

Comment:
Is this a URL or a local app path? Is it possible to pass a string that has the pdb info in it?

Reply:
Do you mean filename? This is already a string e.g. "1BNA.pdb "
It is needed for loading the structure again if it was used before